### DISCUSSION TITLE
repository.deleteById() doesn't work with MongoDB?

### DISCUSSION THREAD

**Discussion by gtamas** [loopbackio/loopback-next#7258] [Q&A] [answered]
Hi

I'm using MongoDB datasource (the latest)

When I try to call `myRepository.deleteById('someObjectId')`, it's unable the find the record.
My repo looks like this:

```
export class UserRepository extends SoftCrudRepository<
  User,
  typeof User.prototype._id,
  UserRelations
  > {
  constructor(
    @inject('datasources.mongo') dataSource: MongoDataSource,
  ) {
    super(User, dataSource);
  }

...
}
```

This is a SoftCrudRepository, but I also tried with DefaultCrudRepository and I'm getting the same result.

My User model has an "_id" field which looks like this:

```
@property({
    type: 'string',
    id: true,
    generated: true,
  })
  _id: string;
```

Should this be called "id"? Or why doesn't this work?

**Comment by gtamas** [comment]
Anybody please?

The all id based methods seem to be having the same issue. I can't find, delete or update by id. EntityNotFound exception is thrown.
I can locate records by any other property, but the id!

I'm sure it's some config problem, but I can't see what's wrong.

**Comment by dougal83** [comment, marked as answer]
> Should this be called "id"? Or why doesn't this work?

With reference to the [mongodb connector docs](https://loopback.io/doc/en/lb4/Connecting-to-MongoDB.html#handling-objectid), try using 'id' as '_id' is reserved.

**Reply by gtamas** [reply]
Yeah, I just figured this out too, but thanks for the info! 
I know now that this is actually mentioned in the loopback docs and it's not only Mongo related. You **have to** call it "id", otherwise the CRUD operations won't work.

But if that's the case, then maybe the docs should updated to make this info easy to find. I mean it's there, but it's a tip, it doesn't really stand out.

Anyway, thanks again! Now it works.

One more thing:

To make it work with Mongo, I had to add this to my model:

```
 @property({
    type: 'string',
    id: true,
    mongodb: {dataType: 'ObjectId'}
  })
  id: string;
```

I added the mongodb prop.
